src/java_bytecode/generic_arguments_name_buildert.h

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************\
+
+ Module: Generic Arguments Name Builder
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+/// file - A class to aid in building the name of specialized generic classes.
+/// Allows for custom builder function for a single argument name.
+
+#ifndef CPROVER_JAVA_BYTECODE_GENERIC_ARGUMENTS_NAME_BUILDERT_H
+#define CPROVER_JAVA_BYTECODE_GENERIC_ARGUMENTS_NAME_BUILDERT_H
+
+#include "java_types.h"
+
+class generic_arguments_name_buildert
+{
+public:
+  generic_arguments_name_buildert(
+    const std::function<std::string(reference_typet)> &argument_name_builder)
```

Comment:
this seems to be problematic for Appveyor, does `#include <functional>` help?

Reply:
Thank you, I will try that.